commons-packet/commons-packet-service/src/main/java/io/mosip/commons/packetmanager/service/PacketReaderService.java

```diff
@@ -1,261 +1,301 @@
+            List<ContainerInfoDto> containerInfoDtos = new ArrayList<>();
+            for (ObjectDto o : allObjects) {
+                if (!containerInfoDtos.stream().anyMatch(info -> info.getSource().equalsIgnoreCase(o.getSource()) && info.getProcess().equalsIgnoreCase(o.getProcess()))) {
+                    ContainerInfoDto containerInfo = new ContainerInfoDto();
+                    containerInfo.setSource(o.getSource());
+                    containerInfo.setProcess(o.getProcess());
+                    containerInfo.setLastModified(o.getLastModified());
+
+                    //get demographic fields
+                    Set<String> demographics = packetReader.getAllKeys(id, containerInfo.getSource(), containerInfo.getProcess());
+                    // get biometrics
+                    List<BiometricsDto> biometrics = null;
+                    BiometricRecord br = packetReader.getBiometric(id, getKey(), Lists.newArrayList(), o.getSource(), o.getProcess(), false);
+                    if (br != null && !CollectionUtils.isEmpty(br.getSegments())) {
+                        Map<String, List<String>> biomap = new HashMap<>();
+                        for (BIR b : br.getSegments()) {
+                            String key = b.getBdbInfo().getType().iterator().next().value();
+                            String subtype = b.getBdbInfo().getSubtype().stream().collect(Collectors.joining(" ")).strip();
+                            if (biomap.get(key) == null)
+                                biomap.put(key, StringUtils.isNotEmpty(subtype) ? Lists.newArrayList(subtype) : null);
+                            else {
+                                List<String> finalVal =  biomap.get(key);
+                                finalVal.add(subtype);
+                                biomap.put(key, finalVal);
+                            }
+                        }
+                        biometrics = new ArrayList<>();
+                        for (Map.Entry<String, List<String>> b : biomap.entrySet()) {
+                            BiometricsDto bioDto = new BiometricsDto();
+                            bioDto.setType(b.getKey());
+                            bioDto.setSubtypes(b.getValue());
+                            biometrics.add(bioDto);
+                        }
+                    }
+
+                    containerInfo.setDemographics(demographics);
+                    containerInfo.setBiometrics(biometrics);
+                    containerInfoDtos.add(containerInfo);
+                }
+            }
+            // get tags
+            Map<String, String> tags = packetReader.getTags(id);
+
+            InfoResponseDto infoResponseDto = new InfoResponseDto();
+            infoResponseDto.setApplicationId(id);
+            infoResponseDto.setPacketId(id);
+            infoResponseDto.setInfo(containerInfoDtos);
+            infoResponseDto.setTags(tags != null ? tags : null);

```

Comment:
this null check is not making much sense

Reply:
Removed null check